azurerm/internal/services/iotcentral/resource_arm_iotcentral_aplication.go

```diff
@@ -0,0 +1,257 @@
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IotCentralAppName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"sub_domain": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.IotCentralAppSubdomain,
+			},
+
+			"display_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.IotCentralAppDisplayName,
+			},
+
+			"sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(iotcentral.S1),
+				}, true),
+				Default: iotcentral.S1,
+			},
+			"template": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmIotCentralAppForCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).IoTCentral.AppsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing IoT Central Application  %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_iotcentral_application", *existing.ID)
+		}
+	}
+
+	req, err := client.CheckNameAvailability(ctx, iotcentral.OperationInputs{
+		Name: utils.String(name),
+	})
+	if err != nil {
+		return fmt.Errorf("Error happend on check name availability. %q (Group Name %q). Error:  %+v", name, resourceGroup, err)
+	}
+	if !*req.NameAvailable {
+		return fmt.Errorf("Resource name not avialable. Reason:  %q, Message  %q", *req.Reason, *req.Message)
+	}
+
+	displayName := d.Get("display_name").(string)
+	if displayName == "" {
+		displayName = name
+	}
+
+	subdomain := d.Get("sub_domain").(string)
+	template := d.Get("template").(string)
+	location := d.Get("location").(string)
+	app := iotcentral.App{
+		AppProperties: &iotcentral.AppProperties{
+			DisplayName: &displayName,
+			Subdomain:   &subdomain,
+			Template:    &template,
+		},
+		Sku: &iotcentral.AppSkuInfo{
+			Name: iotcentral.AppSku(d.Get("sku").(string)),
+		},
+		Location: &location,
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, app)
```

Comment:
is there any separate create function?

Reply:
no, just creatOrUpdate and update functions

Reply:
so we should remove update function. Create and update function should use same function "createOrUpdate", right?